### DISCUSSION TITLE
How do I uninstall this?

### DISCUSSION THREAD

**Discussion by cornerman8** [microsoft/artifacts-credprovider#482] [Q&A]
How do I uninstall this in its totality?

**Comment by embetten** [comment]
Deleting the `CredentialProvider.Microsoft` folder from the netcore and netfx folders under the `.nuget\plugins\` paths will essentially uninstall.

**Reply by TheSmallGrove** [reply]
unfortunately after deleting the path, visual studio 2022 contine to ask for the plugin. Is there a reference to clear?

**Comment by embetten** [comment]
This cred provider is also shipped with visual studio. You could remove it from visual studio by going to the visual studio installer and modifying the installation to remove all of NuGet. 

However, if you are trying to uninstall to ensure you use a different plugin, you could set the %NUGET_CREDENTIALPROVIDERS_PATH% to the tool you prefer (https://learn.microsoft.com/en-us/nuget/reference/extensibility/nuget-exe-credential-providers)

**Reply by TheSmallGrove** [reply]
Removing the env variable solved my problem. Thanks
